File: tests/perf2/test-empty-while-loop.c
/*
 * Empty while loop test.
 */

#include "stdio.h"
#include "time.h"

int main(int argc, char **argv) {
     int msec;
     clock_t start, duration;

     int i = 1e8L;

     start = clock();

     // begin test
     while (--i) ;
     // end test

     duration = clock() - start;
     msec = duration * 1000 / CLOCKS_PER_SEC;

     printf("%d\n", msec);

     return 0;
}

File: tests/perf2/test-empty-while-loop.js
/*
 *   Empty while loop performance
 */

if (typeof window == 'undefined') window = {}
if (!window.performance) window.performance = {}
window.performance.now = window.performance.now || Date.now

if (typeof print != 'undefined') log = print
else log = console.log

function test () {
  var start
  var duration

  var i = 1e8

  start = window.performance.now()

  // begin test
  while (--i) {}
  // end test

  duration = window.performance.now() - start

  log(duration)
}

test()
